src/Sound.cpp

```diff
@@ -1,54 +1,76 @@
 #include "Sound.h"
 
-Sound::Sound(){
+#include <fmod.hpp>
 
-  if( FMOD::System_Create( &m_pSystem) != FMOD_OK){
+#include <iostream>
 
-    std::cout<< "Could not create sound system." <<std::endl;
-    return;
-  }
-
-  int driverCount = 0;
-  m_pSystem->getNumDrivers( &driverCount);
-
-  if( driverCount == 0){
-
-    std::cout<< "no sound driver was found" <<std::endl;
-    return;
-  }
-
-  //system is initiallised with 10 channels for now
-  m_pSystem->init( 10, FMOD_INIT_NORMAL, NULL);
+Sound::Sound() {
+    
+    channel = nullptr;
+    newSound = nullptr;
+    m_pSystem = nullptr;
 }
 
-//loads sound in path to memory
-void Sound::createSound( const char *pFile) {
-
-  m_pSystem->createStream( pFile, FMOD_DEFAULT, 0, &audio);
+Sound::~Sound() {
+    
+    channel->stop();
+    newSound->release();
+    m_pSystem->release();
 }
 
-void Sound::playSound( bool bLoop){
-
-      if ( !bLoop)
-         audio->setMode( FMOD_LOOP_OFF);
-      else{
-
-         audio->setMode( FMOD_LOOP_NORMAL);
-         audio->setLoopCount( -1);
-      }
-
-      playing = m_pSystem->playSound( audio, 0, false, &channel);
-   }
-
-bool Sound::isPlaying(){
+bool Sound::init() {
+    
+    if(FMOD::System_Create(&m_pSystem) != FMOD_OK) {
+
+        std::cout << "Could not create sound system." << std::endl;
+        return false;
+    }
+    
+    int driverCount = 0;
+    m_pSystem->getNumDrivers(&driverCount);
+    
+    if(driverCount == 0) {
+        
+        std::cout << "No sound drivers found!" << std::endl;
+        return false;
+    }
+    
+    //system is initiallised with 10 channels for now
+    m_pSystem->init(10, FMOD_INIT_NORMAL, NULL);
+    
+    return true;
```

Comment:
@ma302jh Since the constructor can't return anything, it's almost useless to try to do any error checking in the constructor.